src/xenia/base/memory.cc

```diff
@@ -34,15 +34,16 @@ void copy_and_swap_16_aligned(void* dest_ptr, const void* src_ptr,
                    0x04, 0x05, 0x02, 0x03, 0x00, 0x01);
 
   size_t i = 0;
-  size_t unaligned_words = (reinterpret_cast<uintptr_t>(src_ptr) & 0xF) / 2;
+  size_t unaligned_words =
+      (0x10 - (reinterpret_cast<uintptr_t>(src_ptr) & 0xF)) / sizeof(uint16_t);
   for (; unaligned_words > 0 && i < count; unaligned_words--, i++) {
     // Copy up to 16 byte alignment.
     dest[i] = byte_swap(src[i]);
   }
   for (; i + 8 <= count; i += 8) {
     __m128i input = _mm_load_si128(reinterpret_cast<const __m128i*>(&src[i]));
     __m128i output = _mm_shuffle_epi8(input, shufmask);
-    _mm_store_si128(reinterpret_cast<__m128i*>(&dest[i]), output);
+    _mm_storeu_si128(reinterpret_cast<__m128i*>(&dest[i]), output);
```

Comment:
We want to use `_mm_store_si128` on aligned pointers.

Reply:
Yeah - I just realized that as well. If both pointers aren't unaligned in the same manner, the `src` pointer will be aligned and the `dest` pointer just gets unaligned.

Do you have a backtrace for those crashes you were getting on linux? I think it's a good idea to assert alignment on both pointers, since this is an aligned routine after all.

Reply:
Adding to that, `src_ptr` and `dst_ptr` should both be aligned (or unaligned by the same amount of bits) or else they only way to avoid a segfault on Linux is to run `_mm_storeu_si128`

Reply:
Here's a backtrace: https://gist.github.com/bwrsandman/7d91bb6a1af79366e15161f992858a09
It happens on the call to `_mm_load_si128`